lib/models/calendar.ts

```diff
@@ -192,7 +211,11 @@ export class Calendar implements BaseCalendar {
           // the baseData must be null.
           const baseData: LiturgicalDay | null = isFromProperOfTime
             ? null
-            : builtData.byKeys[builtData.datesIndex[dateStr][0]];
+            : builtData.byKeys[builtData.datesIndex[dateStr][0]]
+                // Look up for the right LiturgicalDay item, according to its date.
+                // Note: 2 LiturgicalDay objects with the same date can occur within the same liturgical year,
+                // for example, Saint Andrew Apostle (30 November), in 2012.
```

Comment:
> When a same liturgical day can occur twice in the same liturgical year (e.g. Andrew Apostle), the calendar metadata wasn't correct on the first occurrence of this liturgical day

I am curious what do you mean by this quote from the PR description and by this example (L217).

AFAIK, St Andrew is celebrated as _feast_ in the GRC, therefore there cannot be other celebration on that day (unless approved by the Holy See).

Also, how/when/where can St Andrew be celebrated _twice_ in a liturgical year?

Reply:
Good catch. From _2 LiturgicalDay objects with the same **date**_, I wanted to say _...with the same **key**_.

For example, in 2012, the liturgical year starts the 27th November 2011, and ends the 1st December 2012. So within this liturgical year you have twice the 30th November (the day of the feast of St Andrew).
Internally, romcal is computing dates within a liturgical scope, but it has to manage the situation like St Andrew, when civil dates are overlapping.

Reply:
I see. Then let’s be a bit more verbose in the code comment. You might want to change the date format (`30 November 2011` → `November 30, 2011`).

```suggestion
                // for example, Saint Andrew Apostle (30 November 2011 and 30 November 2012), in liturgical year 2012, which starts on 27 November 2011 and ends 1 December 2012.
```